nautobot/docs/plugins/development.md

```diff
@@ -609,6 +610,89 @@ secrets_providers = [ConstantValueSecretsProvider]
+
+Nautobot will dynamically generate the additional relevant lookup expressions of a plugin's defined custom FilterSet field, so no need to additionally register `example_plugin_description__ic`, etc.
+
+Similar to FilterSet fields, Nautobot provides a default filter form for each model, however that does not cover every possible use case. To list a few examples of why one may want to extend a filter form:
+
+* The base filter form does not include a custom filter defined by the plugin as described above
+* The base filter form does not provide a specific lookup expression to a filterable field, such as allowing regex on name
+
+There are several conditions that must be met in order to extend a filter:
+
+* The original FilterForm must follow the pattern: `f"{model.__name__}FilterForm"`, e.g. `TenantFilterForm`
+* The `filterform_fields` attribute must be a valid dictionary of Django form fields
+
+!!! note
+    A plugin is not required to define both `filterset_fields` and `filterform_fields`.
+
+Example:
+```python
+# filter_extensions.py
+from django import forms
+
+from nautobot.extras.plugins import PluginFilterExtension
+from nautobot.utilities.filters import MultiValueCharFilter
+
+
+def suffix_search(queryset, name, value):
+    return queryset.filter(description=f"{value[0]}.nautobot.com")
+
+
+class TenantFilterExtension(PluginFilterExtension):
+    model = "tenancy.tenant"
+
+    filterset_fields = {
+        "example_plugin_description": MultiValueCharFilter(field_name="description", label="Description"),
+        "example_plugin_sdescrip": MultiValueCharFilter(
+            field_name="description", label="Description", method=suffix_search
+        ),
+        "example_plugin_dtype": MultiValueCharFilter(
+            field_name="sites__devices__device_type__slug", label="Device Type"
+        ),
+    }
+
+    filterform_fields = {
+        "example_plugin_description": forms.CharField(required=False, label="Description"),
+        "example_plugin_dtype": forms.CharField(required=False, label="Device Type"),
+        "slug__ic": forms.CharField(required=False, label="Slug Contains"),
+        "example_plugin_sdescrip": forms.CharField(required=False, label="Suffix Description"),
+    }
+
+
+filter_extensions = [TenantFilterExtension]
+```
+
+!!! tip
+    The `method` parameter, if used, must be a callable (method/function). Note that because filters with a `method` do their filtering in Python code rather than at the database level, performance of `method` filters is generally much poorer than pure-database filters. The `method` parameter is not supported when using [dynamic groups](../additional-features/dynamic-groups.md).
```

Comment:
```suggestion
    The `method` parameter, if used, must be a callable (method/function). Note that because filters with a `method` do their filtering in Python code rather than at the database level, performance of `method` filters is generally much poorer than pure-database filters. The `method` parameter is not supported when using [Dynamic Groups](../additional-features/dynamic-groups.md).
```